4-WebApp-your-API/4-1-Your-API/README.md

```diff
@@ -101,368 +79,348 @@ There are two projects in this sample. Each needs to be separately registered in
-string title = string.IsNullOrWhiteSpace(ownerName) ? Todo.Title : $"{Todo.Title} ({ownerName})";
-todoStore.Add(new TodoItem { Owner = owner, Title = title });
-```
+In the TodoListService project, add a folder named `Models` and then create a new  file named `TodoItem.cs`. Copy the contents of the TodoListService\Models\TodoItem.cs in this file.
 
-The work of calling the Microsoft Graph to get the owner name is done in `CallGraphAPIOnBehalfOfUser()`.
+### Modify the startup.cs file to validate bearer access tokens received by the Web API.
 
-This method is the following. It:
+Update `Startup.cs` file :
 
-- gets a token for the Microsoft Graph on behalf of the user (leveraging the token, which was added in the cache on the `TokenValidated` event in `startup.cs`)
-- Calls the graph and retrieves the name of the user.
+- Add the following two using statements
 
-    ```CSharp
-    public async Task<string> CallGraphAPIOnBehalfOfUser()
-    {
-        string[] scopes = new string[] { "user.read" };
-
-        // we use MSAL.NET to get a token to call the API On Behalf Of the current user
-        try
-        {
-            string accessToken = await tokenAcquisition.GetAccessTokenOnBehalfOfUser(HttpContext, scopes);
-            dynamic me = await CallGraphApiOnBehalfOfUser(accessToken);
-            return me.userPrincipalName;
-        }
-        catch (MsalUiRequiredException ex)
-        {
-            tokenAcquisition.ReplyForbiddenWithWwwAuthenticateHeader(HttpContext, scopes, ex);
-            return string.Empty;
-        }
-    }
-    ```
+```CSharp
+using Microsoft.Identity.Web;
+using Microsoft.Identity.Web.Client.TokenCacheProviders;
+```
 
-### Handling required interactions with the user (dynamic consent, MFA, etc ...)
+- In the `ConfigureServices` method, replace the following code:
 
-#### On the Web API side
+  ```CSharp
+  services.AddAuthentication(AzureADDefaults.BearerAuthenticationScheme)
+          .AddAzureADBearer(options => Configuration.Bind("AzureAd", options));
+   ```
 
-An interesting piece is how `MsalUiRequiredException` are handled. These exceptions are typically sent by Azure AD when there is a need for a user interaction. This can be the case when the user needs to re-sign-in, or needs to grant some additional consent, or to obtain additional claims. For instance, the user might need to do multi-factor authentication required specifically by a specific downstream API. When these exceptions happen, given that the Web API does not have any UI, it needs to challenge the client passing all the information enabling this client to handle the interaction with the user.
+  with
 
-This sample uses the `ReplyForbiddenWithWwwAuthenticateHeader` method of the `TokenAcquisition` service. This method uses the HttpResponse to:
+  ```Csharp
+  services.AddProtectWebApiWithMicrosoftIdentityPlatformV2(Configuration)
+          .AddProtectedApiCallsWebApis(Configuration, new string[] { "user.read" })
```

Comment:
`.AddProtectedApiCallsWebApis(Configuration, new string[] { "user.read" })` is not actually used in this sample, so this code snipped should be updated.